ext/odbc/php_odbc.c

```diff
@@ -2093,9 +2124,18 @@ bool odbc_sqlconnect(zval *zv, char *db, char *uid, char *pwd, int cur_opt, bool
 	memset(link->connection, 0, sizeof(odbc_connection));
 
 	link->persistent = persistent;
-	link->hash = hash;
-	SQLAllocEnv(link->connection->henv);
-	SQLAllocConnect(link->connection->henv, link->connection->hdbc);
+	link->hash = zend_string_copy(hash);
+	ret = SQLAllocEnv(&((*link->connection).henv));
```

Comment:
The initial failure which we saw was due to passing `link->connection->henv` instead of `&((*link->connection).henv)` that's why `henv` used to `null` all the time.

Reply:
current master (https://github.com/php/php-src/blob/4a0ec3de468faa0712addda25119ee629d6a44b9/ext/odbc/php_odbc.c#L2026) uses this pattern, so I reverted my accidental change. I'm note sure why someone wrote this 🤔